hr_expense_invoice/models/hr_expense.py

```diff
@@ -38,3 +37,14 @@ def _get_account_move_line_values(self):
                         expense.invoice_id.partner_id.commercial_partner_id.id
                     move_line['account_id'] = expense.invoice_id.account_id.id
         return move_line_values_by_expense
+
+    @api.onchange("invoice_id")
+    def _onchange_invoice_id(self):
+        """Get expense amount from invoice amount. Otherwise it will do a
+           mismatch when trying to post the account move. We also default
+           to the invoice analytic account although it can be changed if
+           needed"""
+        if self.invoice_id:
+            self.quantity = 1
+            self.unit_amount = self.invoice_id.amount_total
+            self.analytic_account_id = self.invoice_id.analytic_account_id
```

Comment:
AFAIK, there's no analytic account at invoice level, but line level.

Reply:
True. I took it for granted... changed

Reply:
Anyway, it should be the same, as this module overrides account.move.line creation for linking the already created ones in invoice, so the analytic account will be used. Am I correct?

Reply:
But we'd have the problem with multiple lines. Wich one has the correct analytic account?

Reply:
It doesn't matter, as the journal items will be generated from the invoice, not from the expense. Take into account that the analytic account goes to the expense account, not to the payable.